src/Sentry.Unity.Editor/AutoInstrumentation/ModuleReadWriter.cs

```diff
@@ -46,15 +132,65 @@ private static bool TryReadSymbols(this ModuleDefinition module)
             }
         }
 
-        public static void Write(StrongNameKeyPair? key, bool hasSymbols, ModuleDefinition module, string file)
+        internal TypeDefinition GetTypeDefinition(ModuleDefinition module, Type type)
         {
-            var parameters = new WriterParameters
+            var typeDefinition = module.GetType(type.FullName);
+            if (typeDefinition is null)
             {
-                WriteSymbols = hasSymbols,
-                StrongNameKeyPair = key
-            };
+                throw new ArgumentException($"Failed to get requested type definition in {module.Name}", type.FullName);
+            }
+
+            return typeDefinition;
+        }
+
+        internal MethodDefinition GetMethodDefinition(TypeDefinition typeDefinition, string name, Type[]? requiredParameters = null)
+        {
+            foreach (var method in typeDefinition.Methods)
+            {
+                if (method?.Name == name)
+                {
+                    switch (requiredParameters)
+                    {
+                        case null when !method.HasParameters:
+                            return method;
+                        case null:
+                            continue;
+                    }
+
+                    if (method.Parameters.Count != requiredParameters.Length)
+                    {
+                        continue;
+                    }
+
+                    // We go over all the required parameters and compare them to the parameters found in the method.
```

Comment:
Is this logic correct? It means that if each of the required parameters is found *anywhere* in the method signature, then it would match.

For example, `Foo(string, string, int, int)` would also match `Foo(int, int, string, string)`

Reply:
See my suggested change below also.  It's a bit easier to see in LINQ.